### PR TITLE
Solution for day 21

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

File: src/day21.scala [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    for
+      die1 <- List(1, 2, 3)
+      die2 <- List(1, 2, 3)
+      die3 <- List(1, 2, 3)
+    yield
+      die1 + die2 + die3
+  possibleRolls.groupMapReduce(identity)(_ => 1L)(_ + _).toList
+
+def playWithDiracDie(players: Players, player1Turn: Boolean, wins: Wins, inHowManyUniverses: Long): Unit =
+  for (diesValue, count) <- dieCombinations yield
```
Comment (julienrf) [inline src/day21.scala:83]: Why bother with `yield`ing something in the absence of comonad?

Comment (sjrd) [reply src/day21.scala]: Oops, leftover from when I thought the functional way was going to be easier.

File: src/day21.scala [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    players(1).score
+  else
+    val newPlayer = Player(newCell, newScore)
+    playWithDeterministicDie((players(1), newPlayer), die)
+
+final class Wins(var player1Wins: Long, var player2Wins: Long)
+
+def part2(input: String): Long =
+  val players = parseInput(input)
+  val wins = new Wins(0L, 0L)
```
Comment (julienrf) [inline src/day21.scala:67]: Not really important, but `new` is not necessary.

**Review by julienrf — APPROVED** [review]

File: src/day21.scala (hunk 2 above)
Comment (sjrd) [reply src/day21.scala:67]: I like using `new` explicitly when I create stateful instances.

**Pull request merged**